tsl/src/nodes/decompress_chunk/decompress_chunk.c

```diff
@@ -1839,11 +1841,31 @@ create_compressed_scan_paths(PlannerInfo *root, RelOptInfo *compressed_rel, Comp
 		 * decompression
 		 */
 		List *orig_pathkeys = root->query_pathkeys;
+		List *orig_eq_classes = root->eq_classes;
 		build_compressed_scan_pathkeys(sort_info, root, root->query_pathkeys, info);
 		root->query_pathkeys = sort_info->required_compressed_pathkeys;
+
+		/* We can optimize iterating over EquivalenceClasses by reducing them to
+		 * the subset which are from the compressed chunk. This only works if we don't
+		 * have joins involved since those use eclass_indexes which is not valid
+		 * with this optimization.
+		 */
+		if (info->chunk_rel->joininfo == NIL && !info->chunk_rel->has_eclass_joins)
```

Comment:
Does this work for a clauseless inner join, e.g. `select from table1, table2 order by something`? I think both these conditions will be false in this case. Might be good to add a test case for this.

Reply:
So in the end, what happens for the clauseless join? It goes through to `generate_join_implied_equalities`, but since there are no common equivalence classes, it doesn't matter that the `eclass_indexes` are empty, so it works fine in the end? Let's mention this in the comment.

I also wonder if we have to test for `joininfo`, these are the non-EC join clauses (e.g. a join on inequality), probably they don't use the `eclass_indexes` and it will work fine the same way as the clauseless?

Reply:
I think you are right about `joininfo`, lets remove that check.